### PR TITLE
Re-write examples

### LINKED ISSUE #450: Clean up examples
## 📚 Documentation

We should standardize the examples:

- example template
- each example should be:
    - imports
    - datamodule
    - task
    - trainer / train / finetune
    - predict
    - save the model
- as minimal as possible (but showing e.g. selecting backbones)
- merge finetune & predict examples into the above
- move `integrations/fiftyone` -> `visualization` and add other visualization examples

cc @aribornstein

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Fixes #450 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — COMMENTED** [review]
seems like ground-up rewrite

File: docs/source/_templates/theme_variables.jinja [hunk 1]
```diff
@@ -14,5 +14,7 @@
   'blog': 'https://www.pytorchlightning.ai/blog',
   'resources': 'https://lightning-flash.readthedocs.io/',
   'support': 'https://lightning-flash.readthedocs.io/',
+  'community': 'https://pytorch-lightning.slack.com',
+  'forums': 'https://pytorch-lightning.slack.com',
```
Comment (akihironitta) [inline docs/source/_templates/theme_variables.jinja:18]: Maybe the invitation link here? New people cannot get to the workspace with the current link.
```suggestion
  'community': 'https://pytorch-lightning.slack.com/join/shared_invite/zt-pw5v393p-qRaDgEk24~EjiZNBpSQFgQ',
  'forums': 'https://pytorch-lightning.slack.com/join/shared_invite/zt-pw5v393p-qRaDgEk24~EjiZNBpSQFgQ',
```

Comment (ethanwharris) [reply docs/source/_templates/theme_variables.jinja:18]: @akihironitta Yeah, although I can't see where this link actually appears in the docs, so maybe doesn't matter haha

**Review by tchaton — APPROVED** [review]

File: docs/source/_templates/layout.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{% extends "!layout.html" %}
+<link rel="canonical" href="{{ theme_canonical_url }}{{ pagename }}.html" />
+
+{% block footer %}
+{{ super() }}
+<script script type="text/javascript">
+  var collapsedSections = ['Guides', 'Image', 'Tabular', 'Text', 'Video', 'Integrations', 'API Reference', 'Contributing a Task'];
```
[the remark this answers is not in the document]
Comment (ethanwharris) [reply docs/source/_templates/layout.html]: Done :smiley:

**Pull request merged**